Bio/Seq.py

```diff
@@ -2052,6 +2052,22 @@ def ungap(self, gap="-"):
             raise ValueError(f"Unexpected gap character, {gap!r}")
         return self.replace(gap, b"")
 
+    @property
+    def defined(self):
+        """Return true if the sequence is defined, false if undefined or partially defined.
+
+        Zero-length sequences are always considered to be defined.
+        """
+        if len(self) == 0:
+            return True
+        elif isinstance(self, UnknownSeq):
+            return False
+        elif isinstance(
+            self._data, (_UndefinedSequenceData, _PartiallyDefinedSequenceData)
+        ):
+            return False
```

Comment:
I would prefer:
```python
if isinstance(self._data, (bytes, bytearray)):
    return True
else:
    return self._data.defined
```
and add a `defined` attribute to `SequenceDataAbstractBaseClass` that always returns `True`, and a `defined` attribute to `_UndefinedSequenceData` and `_PartiallyDefinedSequenceData` that always returns `False`.